### ISSUE TITLE
Editor scene tree navigation and selection using keyboard is not intuitive

### ISSUE THREAD

**Issue by doongjohn** [godotengine/godot#74098]
### Godot version

4.0 rc6

### System information

Windows 11

### Issue description

1. Scene tree area loses focus when you hover the mouse to the scene(game) view.
2. Navigating with arrow key does not highlight selected node and gizmo is not shown for that node.
3. Shift + arrow does not select multiple nodes.
![Recording 2023-02-28 at 14 43 55](https://user-images.githubusercontent.com/40219740/221764876-68331775-1997-43fe-85bf-74aa37fe1d37.gif)

### Steps to reproduce

1. Scene tree area loses focus when you hover the mouse to the scene view.
  - Select scene tree view with the mouse.
  - Hover over the scene view.
  - It loses focus so you can't use arrow keys to navigate scene tree.
2. Navigating with arrow key does not highlight selected node and gizmo is not shown for that node.
  - Selected node using mouse has a blue background and the transform gizmo is shown in the scene view.
  - If you use arrow key to select a node in a scene tree, it has no blue background and the gizmo of the node you previously selected with the _mouse_ is still shown.
3. Shift + arrow does not select multiple nodes.
  - Shift + mouse click can select multiple nodes.
  - Shift + arrow up / down does not select multiple nodes. In fact it does nothing.

### Minimal reproduction project

N/A

**Comment by doongjohn** [comment]
Similar issue also applies to the File system area.

**Comment by EntroPiGames** [comment]
> 2\. Navigating with the arrow keys do not highlight selected node and gizmo is not shown for that node.

I agree that this is very counter intuitive, especially coming from Unity or Blender. It also makes navigating scenes using the Scene Tree dock (which I often do in larger scenes) rather impractical.

Adding an editor input action to select the highlighted Node would be a great first improvement.

**Comment by hsandt** [comment]
> Adding an editor input action to select the highlighted Node would be a great first improvement. Pressing enter in the `File System` dock already selects and opens the selected resource, but in the `Scene` dock pressing enter renames the Node.

Exactly, and both Space and F2 does the same as Enter: renames, but doesn't even select the node. Space and/or Enter could be used for selection, and F2 for rename anyway.

**Comment by hsandt** [comment]
Interestingly, when debugging the Remote tree works exactly as we want, moving the selection completely (except F2 can't rename).

![Godot 4 2 1 - Remote tree correct navigation](https://github.com/godotengine/godot/assets/1067772/aa0285d7-8d0a-489f-85ec-7563a4d005e0)
